src/ObjectFactory.php

```diff
@@ -58,6 +56,7 @@ public function create(callable|array $attributes = []): object
 
         $parameters = $this->normalizeParameters($parameters);
         $instantiator = $this->instantiator ?? Configuration::instance()->instantiator;
+        /** @var T $object */
```

Comment:
it seems like PHPStorm does not only interprets methods signature (like PHPStan would do) but it events guesses stuff from the methods' body. I had really hard times to find but it basically fixes 90% of auto-complete problems :tada: